src/srdf_writer.cpp

```diff
@@ -315,32 +316,71 @@ void SRDFWriter::createLinkSphereApproximationsXML(XMLElement* root)
 }
 
 // ******************************************************************************************
-// Generate XML for SRDF disabled collisions of robot link pairs
+// Generate XML for SRDF collision defaults of robot links
 // ******************************************************************************************
-void SRDFWriter::createDisabledCollisionsXML(XMLElement* root)
+void SRDFWriter::createCollisionDefaultsXML(XMLElement* root)
 {
   XMLDocument* doc = root->GetDocument();
 
   // Convenience comments
-  if (disabled_collisions_.size())  // only show comments if there are corresponding elements
+  if (!no_default_collision_links_.empty())  // only show comments if there are corresponding elements
   {
-    XMLComment* comment = doc->NewComment("DISABLE COLLISIONS: By default it is assumed that any link of the robot "
+    XMLComment* comment = doc->NewComment("DEFAULT COLLISIONS: By default it is assumed that any link of the robot "
                                           "could potentially come into collision with any other link in the robot. "
-                                          "This tag disables collision checking between a specified pair of links. ");
+                                          "This tag allows to revert this behavior and disable collisions by default.");
     root->InsertEndChild(comment);
   }
 
-  for (std::vector<srdf::Model::DisabledCollision>::const_iterator pair_it = disabled_collisions_.begin();
-       pair_it != disabled_collisions_.end(); ++pair_it)
+  for (const std::string& name : no_default_collision_links_)
   {
     // Create new element for each link pair
-    XMLElement* link_pair = doc->NewElement("disable_collisions");
-    link_pair->SetAttribute("link1", pair_it->link1_.c_str());
-    link_pair->SetAttribute("link2", pair_it->link2_.c_str());
-    link_pair->SetAttribute("reason", pair_it->reason_.c_str());
+    XMLElement* entry = doc->NewElement("collision_default");
+    entry->SetAttribute("link", name.c_str());
+    entry->SetAttribute("allow", "ALWAYS");
+    root->InsertEndChild(entry);
+  }
+  // Write enabled collision pairs
+  createCollisionPairsXML(root, "enable_collisions", false);
```

Comment:
It seems very weird to call this method here instead of at the end of `createCollisionPairsXML` below.

Reply:
Logically, re-enabling collisions is linked to the function `createCollisionDefaultsXML()` here, isn't it?
The method `createCollisionPairsXML(XMLElement*)` refers to the old method called to disable collision pairs.

Reply:
No hard requirement from my side to pass review. I would just not expect something called "create collision defaults" to generate collision **pairs** as well.
I would expect that to be done by calling "create collision pairs" instead :cat2: 
Otherwise you might want to consider renaming methods to reflect this better.

Reply:
I want to keep it here.